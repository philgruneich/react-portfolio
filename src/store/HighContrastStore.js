import React, { useState, createContext } from 'react';

export const HighContrastContext = createContext();
HighContrastContext.displayName = 'HighContrastContext';

export const HighContrastStore = props => {
  const [hasContrast, saveHasContrast] = useState(false);

  const toggleContrast = () => {
    saveHasContrast(() => !hasContrast);
  }

  return (
    <HighContrastContext.Provider
      value={{
        state: hasContrast,
        dispatch: saveHasContrast,
        toggle: toggleContrast
      }}
      {...props}
    />
  )
}
