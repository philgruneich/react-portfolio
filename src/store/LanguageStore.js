import React, { useState, useEffect, createContext } from 'react';
import { languages } from '~/lang';

export const LanguageContext = createContext();
LanguageContext.displayName = 'LanguageContext';

export const LanguageStore = props => {
  const [lang, saveLang] = useState('en');
  const [translations, saveTranslations] = useState(languages[lang]);

  useEffect(() => {
    saveTranslations(languages[lang]);
  }, [lang]);

  return (
    <LanguageContext.Provider
      value={{
        state: { lang, translations },
        dispatch: saveLang
      }}
      {...props}
    />
  )
}
