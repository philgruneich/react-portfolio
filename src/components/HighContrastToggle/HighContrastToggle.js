import React, { useContext } from 'react';
import { HighContrastContext } from '~/store/HighContrastStore';
import styled from 'styled-components';

export const HighContrastToggle = () => {
  const contrastContext = useContext(HighContrastContext);

  const onToggleHandler = event => {
    event.preventDefault();

    contrastContext.toggle();
  }

  return (
    <HighContrastButton
      isActive={contrastContext.state}
      aria-pressed={contrastContext.state}
      onClick={onToggleHandler}
    >
      <svg width="26" height="16" viewBox="0 0 26 16" xmlns="http://www.w3.org/2000/svg"><g fill="currentColor"><path d="M13.09 0c5.086.072 10.754 2.187 12.493 6.575 1.391 3.51-3.194 7.193-6.96 8.473-.816.277-1.655.488-2.504.636C10.35 16.687 3.265 15.382.44 9.872-1.9 5.306 5.491-.036 13.09 0zm-.36 2C8.563 2.06 3.844 3.57 2.165 7.104c-1.516 3.192 5.457 6.85 10.734 6.85.913 0 1.826-.068 2.727-.216 3.404-.557 7.042-2.437 8.097-5.096C24.951 5.545 18.421 1.974 12.73 2z" /><path d="M13.028 2.978c4.308.165 7.009 7.409 1.818 9.606C11.279 14.093 6.373 10 8.394 5.809c.843-1.75 2.418-2.858 4.634-2.83zm-.206 2c-2.604.099-4.177 4.457-1.089 5.763 1.93.817 4.53-1 4.12-3.295-.254-1.422-1.431-2.488-3.031-2.468z" /></g></svg>
    </HighContrastButton>
  )
}

const HighContrastButton = styled.button.attrs({
  type: 'button',
  'aria-label': 'Alto Contraste'
})`
  appearance: none;
  margin: 0;
  padding: 0;
  border: 0;
  border-radius: 0;
  background: none;
  color: #fff;
  cursor: pointer;

  ${props => props.isActive && `
    color: green;
  `}
`;
