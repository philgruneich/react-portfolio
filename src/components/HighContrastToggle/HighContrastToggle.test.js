import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { HighContrastStore } from '~/store/HighContrastStore';
import { HighContrastToggle } from './HighContrastToggle';

const defaultHighContrastState = {
  state: false
}

test('Matches snapshot', () => {
  const { asFragment } = render(
    <HighContrastStore value={defaultHighContrastState}>
      <HighContrastToggle />
    </HighContrastStore>
  );

  expect(asFragment()).toMatchSnapshot();
})

test('Matches snapshot with high contrast', () => {
  const { asFragment } = render(
    <HighContrastStore value={{
      state: true
    }}>
      <HighContrastToggle />
    </HighContrastStore>
  );

  expect(asFragment()).toMatchSnapshot();
});

test('Clicking button toggles high contrast', () => {
  const toggleFn = jest.fn();
  const { getByLabelText } = render(
    <HighContrastStore value={{
      state: true,
      toggle: toggleFn
    }}>
      <HighContrastToggle />
    </HighContrastStore>
  );

  const button = getByLabelText('Alto Contraste');

  fireEvent.click(button);

  expect(toggleFn).toBeCalled();
});
