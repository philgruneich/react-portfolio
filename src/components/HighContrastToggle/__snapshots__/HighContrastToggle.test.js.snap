// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Alto Contraste"
    aria-pressed="false"
    class="HighContrastToggle__HighContrastButton-dyr20h-0 hVpOTY"
    type="button"
  >
    <svg
      height="16"
      viewBox="0 0 26 16"
      width="26"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="currentColor"
      >
        <path
          d="M13.09 0c5.086.072 10.754 2.187 12.493 6.575 1.391 3.51-3.194 7.193-6.96 8.473-.816.277-1.655.488-2.504.636C10.35 16.687 3.265 15.382.44 9.872-1.9 5.306 5.491-.036 13.09 0zm-.36 2C8.563 2.06 3.844 3.57 2.165 7.104c-1.516 3.192 5.457 6.85 10.734 6.85.913 0 1.826-.068 2.727-.216 3.404-.557 7.042-2.437 8.097-5.096C24.951 5.545 18.421 1.974 12.73 2z"
        />
        <path
          d="M13.028 2.978c4.308.165 7.009 7.409 1.818 9.606C11.279 14.093 6.373 10 8.394 5.809c.843-1.75 2.418-2.858 4.634-2.83zm-.206 2c-2.604.099-4.177 4.457-1.089 5.763 1.93.817 4.53-1 4.12-3.295-.254-1.422-1.431-2.488-3.031-2.468z"
        />
      </g>
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Matches snapshot with high contrast 1`] = `
<DocumentFragment>
  <button
    aria-label="Alto Contraste"
    aria-pressed="true"
    class="HighContrastToggle__HighContrastButton-dyr20h-0 cvjVRL"
    type="button"
  >
    <svg
      height="16"
      viewBox="0 0 26 16"
      width="26"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="currentColor"
      >
        <path
          d="M13.09 0c5.086.072 10.754 2.187 12.493 6.575 1.391 3.51-3.194 7.193-6.96 8.473-.816.277-1.655.488-2.504.636C10.35 16.687 3.265 15.382.44 9.872-1.9 5.306 5.491-.036 13.09 0zm-.36 2C8.563 2.06 3.844 3.57 2.165 7.104c-1.516 3.192 5.457 6.85 10.734 6.85.913 0 1.826-.068 2.727-.216 3.404-.557 7.042-2.437 8.097-5.096C24.951 5.545 18.421 1.974 12.73 2z"
        />
        <path
          d="M13.028 2.978c4.308.165 7.009 7.409 1.818 9.606C11.279 14.093 6.373 10 8.394 5.809c.843-1.75 2.418-2.858 4.634-2.83zm-.206 2c-2.604.099-4.177 4.457-1.089 5.763 1.93.817 4.53-1 4.12-3.295-.254-1.422-1.431-2.488-3.031-2.468z"
        />
      </g>
    </svg>
  </button>
</DocumentFragment>
`;
